src/decision_tree.cpp

```diff
@@ -260,37 +260,30 @@ std::pair<int,double> DecisionTree::findBestSplit(TreeNode *node) const
     // Must have enough data to split
     assert (dataframe.length()>1);
     std::pair<int,double> split;
-    // How many features to try.
-    int m; 
     // Vector of indices which may or may not be shuffled.
-    std::vector<int> shuf_inds(this->num_features_); 
-    if (this->mtry_==-1){
-        // Deterministic:
-        m = this->num_features_;
-        // Add indices without actually shuffling:
-        for (int i = 0; i < this->num_features_; i++){
-            shuf_inds[i] = i;
-        }
+    std::vector<int> shuf_inds(this->num_features_);
+    // Shuffle if mtry_ < num_features_ else deterministic
+    if (this->mtry_ < this->num_features_) {
+      // Create vector of column indices, equivalent to np.arange(0, df.shape[-1])
+      std::generate(shuf_inds.begin(), shuf_inds.end(), [n = 0] () mutable { return n++; });
```

Comment:
Could actually put this outside the `if` and do nothing with it unless the `if`-condition is met (which then only shuffles it).

Reply:
Great, updated